viewer.py

```diff
@@ -116,33 +118,68 @@ def send_current_asset_id_to_server():
     consumer.send({'current_asset_id': scheduler.current_asset_id})
 
 
+def setup_wifi(data):
+    if not is_balena_app():
+        return
+
+    uri = 'http://{0}/hotspot'.format(LISTEN)
+    sleep_duration = 60
+
+    decoded = json.loads(data)
+
+    # @TODO: Refactor HTML file generation into a separate function.
+    base_dir = path.abspath(path.dirname(__file__))
+    template_path = path.join('templates/hotspot.html')
+
+    with open(template_path) as f:
+        template = Template(f.read())
+
+    context = {
+        'network': decoded.get('network', None),
+        'ssid_pswd': decoded.get('ssid_pswd', None),
+        'address': decoded.get('address', None),
+    }
+
+    with open('/tmp/hotspot.html', 'w') as out_file:
+        out_file.write(template.render(context=context))
```

Comment:
@vpetersson @rusko124, I might need NGINX-related help on this one. I've managed to have `hotspot.html` written in `/tmp`, but what's left to do (apart from the license compatibility stuff) is to fix `nginx.conf`.

In the meantime, I'll read up (and/or do a very quick crash course) on writing NGINX config files. While the syntax might seem straightforward, there could be some catches.